website_mass_mailing_name/static/src/js/website_mass_mailing_name.js

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2016-2017 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+
+odoo.define("website_mass_mailing_name.subscribe", function (require) {
+    "use strict";
+    require("mass_mailing.website_integration");
+    var animation = require("website.content.snippets.animation");
+
+    animation.registry.subscribe.include({
+        start: function(editable_mode) {
+            this.$email = this.$(".js_subscribe_email");
+            this.$name = this.$(".js_subscribe_name");
+            // Thanks upstream for your @$&#?!! inheritance-ready code.
+            // Injecting ajax events to modify behavior of snippet.
+            if (this.$name) {
+                $(document).ajaxSend($.proxy(this.on_ajax_send, this));
+            }
+            return this._super(editable_mode);
+        },
+
+        _onClick: function() {
+            function validateEmail(sEmail) {
+                var filter = /^([\w-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/;
+                if (filter.test(sEmail)) {
+                    return true;
+                }
+                else {
+                    return false;
+                }
+            }
+            var email_error = validateEmail(!this.$email.val()),
```

Comment:
```suggestion
            var email_error = !this.$email[0].checkValidity(),
```

This would still be a false positive if the email is empty. See below on how to fix that.